lib/redis/namespace.rb

```diff
@@ -364,6 +364,15 @@ def call_with_namespace(command, *args, &block)
 
       (before, after) = handling
 
+      # Avoid modifying the caller's arguments.
+      args = args.map do |arg|
```

Comment:
I'm not sure I fully follow - it's the hashes that are being modified. `*args` is a list that is local, but the hash is shared with the caller:

```